python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -0,0 +1,808 @@
+
+    return LiteBlend(sources=sources, observation=None)
+
+
+def scarletLiteToData(blend, psfCenter, xy0):
+    """Convert a scarlet lite blend into a persistable data object
+
+    Parameters
+    ----------
+    blend : `scarlet.lite.LiteBlend`
+        The blend that is being persisted.
+    psfCenter : `tuple` of `int`
+        The center of the PSF.
+    xy0 : `tuple` of `int`
+        The lower coordinate of the entire blend.
+
+    Returns
+    -------
+    blendData : `ScarletBlendDataModel`
+        The data model for a single blend.
+    """
+    sources = {}
+    for source in blend.sources:
+        components = []
+        for component in source.components:
+            if isinstance(component, LiteFactorizedComponent):
+                componentData = ScarletFactorizedComponentData(
+                    xy0=tuple(int(x) for x in component.bbox.origin[1:][::-1]),
+                    extent=tuple(int(x) for x in component.bbox.shape[1:][::-1]),
+                    center=tuple(int(x) for x in component.center[::-1]),
+                    sed=component.sed,
+                    morph=component.morph,
+                )
+            else:
+                componentData = ScarletComponentData(
+                    xy0=tuple(int(x) for x in component.bbox.origin[1:][::-1]),
+                    extent=tuple(int(x) for x in component.bbox.shape[1:][::-1]),
+                    center=tuple(int(x) for x in component.center[::-1]),
+                    model=component.get_model(),
+                )
+            components.append(componentData)
+        sourceData = ScarletSourceData(
+            modelType=source.__class__.__name__,
+            components=[],
+            factorizedComponents=components,
+            peakId=source.peakId,
+        )
+        sources[source.recordId] = sourceData
+
+    blendData = ScarletBlendData(
+        xy0=(xy0.x, xy0.y),
+        extent=blend.observation.bbox.shape[1:][::-1],
+        sources=sources,
+        psfCenter=psfCenter,
+    )
+
+    return blendData
+
+
+def scarletToData(blend, psfCenter, xy0):
+    """Convert a scarlet blend into a persistable data object
+
+    Parameters
+    ----------
+    blend : `scarlet.Blend`
+        The blend that is being persisted.
+    psfCenter : `tuple` of `int`
+        The center of the PSF.
+    xy0 : `tuple` of `int`
+        The lower coordinate of the entire blend.
+
+    Returns
+    -------
+    blendData : `ScarletBlendDataModel`
+        The data model for a single blend.
+    """
+    sources = {}
+    for source in blend.sources:
+        componentData = ScarletComponentData(
+            xy0=tuple(int(x) for x in source.bbox.origin[1:][::-1]),
+            extent=tuple(int(x) for x in source.bbox.shape[1:][::-1]),
+            center=tuple(int(x) for x in source.center[::-1]),
+            model=source.get_model(),
+        )
+
+        sourceData = ScarletSourceData(
+            modelType=source.__class__.__name__,
```

Comment:
Ah, so fortunately the name is just meta data and not actually used in reconstruction. In scarlet lite there is (and should always be) only one type of source class, whereas that was a much more important thing to know in scarlet main. Since this is already stored in the `deblendedCatalg` (https://github.com/lsst/meas_extensions_scarlet/blob/main/python/lsst/meas/extensions/scarlet/scarletDeblendTask.py#L1450) I can just remove this altogether.